### DISCUSSION TITLE
How to run Aspire dashboard in private azure container apps environment?

### DISCUSSION THREAD

**Discussion by JasonRodman** [microsoft/aspire#6960] [Q&A]
We have several apps setup and deployed to ACA with aspire configured locally and were looking for a way to use the aspire dashboard there as well. We were excited to find they had recently added an option to enable it, but were quickly disappointed when we found it didn't work. We run our ACA environment with private endpoints so it's only accessible inside our network. Once you make that switch, the auto-generated URLs for all container apps do not work. It appears azure does not add the dns entries so they do not resolve. We worked around this with our own private dns zone and mapped custom domains to every container that needed an ingress. Now comes along aspire and they only way to access the dashboard is thru the auto-generated url which doesn't work. We are stuck. 

To work around this, we are experimenting with running the aspire dashboard as a standalone container app. This is probably what azure is doing internally anyway, and it works fine once we map a custom domain to it. The challenge we face now is how do we get our telemetry to it? There is no documentation anywhere on how the aspire dashboard gets it data. We know that the other containers are not sending their telemetry directly to the aspire dashboard, they send it to the shared OTEL collector at http://k8se-otel.k8se-apps.svc.cluster.local:4317. So we figured maybe we just have to point the aspire dashboard so it can "pull" the data from there. We set the "Dashboard:ResourceServiceClient:Url" config setting by overriding it with an environment variable and the dashboard fails to start because the auth settings are missing. We then set the "Dashboard:ResourceServiceClient:AuthMode" setting to unsecured and the dashboard starts, but the client has errors on a socket with no detail. I assume it can't read the telemetry data from the collector. 

Next we tried adding an OTEL endpoint in aca that points directly to the aspire dashboard, under the assumption that the otel collector would forward telemetry to it. Nothing ever shows up in the dashboard, and no visibility into why. Also, we found that some of the features disappear from the dashboard when using this approach, so we figured this probably is not the right way. 

At this point we are stuck. We have an open ticket with Microsoft about the challenge with viewing the built-in dashboard without custom domain support, but I am hoping you guys can help give me some guidance on how to setup the dashboard manually as a workaround for now.

**Comment by CRidge** [comment]
We have several internal ACAs set up with a reverse proxy in front of it. It would be really great if there was a way to route to the dashboard from externally, but I can't seem to get it to work - even with the preview http route config feature. It would be great if someone from "the inside" either could point us in the right direction - or confirm it isn't possible.

**Comment by davidfowl** [comment]
cc @snehapar9

**Comment by mcrio** [comment]
I was pleasantly surprised to see the Aspire Dashboard in ACA. That's a killer feature! But still, what's the recommended way accessing the dashboard when ACA is internal to the assigned VNet?
